core/src/ru/spbau/blackout/GameWorld.java

```diff
@@ -59,7 +65,76 @@ public GameWorld() {
         return gameObjects.iterator();
     }
 
-    public void update(float delta) {
+    @Override
+    public synchronized void inplaceSerialize(ObjectOutputStream out) throws IOException, ClassNotFoundException {
+        out.writeInt(gameObjects.size());
+
+        for (GameObject object : this) {
+            //System.out.println("some object des");
+            //object.inplaceSerialize(out);
+
+            Vector2 sending = object.getPosition();
+            out.writeFloat(sending.x);
+            out.writeFloat(sending.y);
+            if (object instanceof GameUnit) {
+                GameUnit unit = (GameUnit) object;
+                sending = unit.getSelfVelocity();
+                out.writeFloat(sending.x);
+                out.writeFloat(sending.y);
+            }
+
+            // ???? do not work
```

Comment:
Question actually lies here. If we print Vector2 right from the object then we will get the same results not matter how game is progressing or how much time has passed since the start of the game on Client Thread. But if we copy the values before sending them then we will get wanted results.
It seems like the problem with multithreading and more specifically with thread's local cache.